### PR TITLE
feat(bucket): allow user to clear histories of each bucket

### PR DESCRIPTION
**Pull request by htuna07**

### CODE REVIEW

File: stacks/api/bucket/history/history.controller.spec.ts [hunk 1]
```diff
@@ -211,4 +220,20 @@ describe("History Acceptance", () => {
       description: "current description"
     });
   });
+  it("should clear histories for given bucketId", async () => {
+    const response = await req.delete(`/bucket/${bucketId}/history`);
+    expect(response.statusCode).toEqual(204);
+    expect(response.body).toEqual(undefined);
+
+    const {body: deletedHistories} = await req.get(`/bucket/${bucketId}/history/${documentId}`, {});
+    expect(deletedHistories).toEqual([]);
```
Comment (thesayyn) [inline stacks/api/bucket/history/history.controller.spec.ts:229]: The body should be undefined.

File: stacks/spica/src/bucket/pages/bucket-add/bucket-add.component.html [hunk 2]
```diff
@@ -405,4 +405,29 @@ <h4 mat-subheader><mat-icon>view_module</mat-icon> View</h4>
...
+      <mat-progress-spinner
+        *ngIf="isHistoryClearPending"
+        [diameter]="18"
+        color="primary"
+        mode="indeterminate"
+      ></mat-progress-spinner>
+      <span>
+        {{ isHistoryClearPending ? "Removing" : "Remove" }}
+      </span>
+    </button>
```
Comment (thesayyn) [inline stacks/spica/src/bucket/pages/bucket-add/bucket-add.component.html:428]: Could you make use of matSave directive here?

File: stacks/api/bucket/history/history.controller.spec.ts (hunk 1 above)
Comment (htuna07) [reply stacks/api/bucket/history/history.controller.spec.ts:229]: Shouldn't be an empty array if there isn't any history? Also, I didn't make any changes to this endpoint logic.

File: stacks/spica/src/bucket/pages/bucket-add/bucket-add.component.ts [hunk 3]
```diff
@@ -50,6 +51,10 @@ export class BucketAddComponent implements OnInit, OnDestroy {
 
   $save: Observable<SavingState>;
 
+  $remove: Observable<SavingState>;
+
+  isHistoryClearPending = false;
```
Comment (thesayyn) [inline stacks/spica/src/bucket/pages/bucket-add/bucket-add.component.ts:56]: You have got an unneeded statement here.

File: stacks/spica/src/bucket/pages/bucket-add/bucket-add.component.html [hunk 4]
```diff
@@ -405,4 +405,42 @@ <h4 mat-subheader><mat-icon>view_module</mat-icon> View</h4>
...
+        ><span>Removing..</span>
+      </div>
+      <div *ngSwitchCase="'saved'">
+        <mat-icon>done</mat-icon>
+        <span>Removed</span>
+      </div>
+      <div *ngSwitchCase="'failed'">
+        <mat-icon>clear</mat-icon>
+        <span>Failed</span>
+      </div>
```
[the remark this answers is not in the document]
Comment (htuna07) [reply stacks/spica/src/bucket/pages/bucket-add/bucket-add.component.html:440]: Is it possible to give style to ng-container? If it's possible, do you know how?

File: stacks/spica/src/bucket/pages/bucket-add/bucket-add.component.html [hunk 5]
```diff
@@ -405,4 +405,42 @@ <h4 mat-subheader><mat-icon>view_module</mat-icon> View</h4>
...
+    <mat-divider></mat-divider>
+    <h3>
+      DANGER ZONE
+    </h3>
+    <button
+      (click)="clearHistories()"
+      mat-raised-button
+      color="warn"
+      *matSave="$remove | async; let state"
+      [disabled]="!!state"
```
[the remark this answers is not in the document]
Comment (htuna07) [reply stacks/spica/src/bucket/pages/bucket-add/bucket-add.component.html:418]: It's not working when I make this change. It stays always disabled.

**Pull request merged**